btree/btree.c

```diff
@@ -3995,11 +4012,15 @@ int64_t btree_iter_tick(struct m0_sm_op *smop)
 			node_rec(&s);
 		}
 		bop->bo_cb.c_act(&bop->bo_cb, &s.s_rec);
-		lock_op_unlock(tree);
+		lock_op_unlock(tree, oi);
 		return m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_DONE);
 	}
 	case P_CLEANUP:
 		level_cleanup(oi, bop->bo_tx);
+		return P_FINI;
+	case P_FINI:
+		if(oi)
```

Comment:
Remove if check. The code should make sure that we do not reach this state if memory is not allocated. If you want to catch bugs for this case then add M0_ASSERT.

Reply:
done